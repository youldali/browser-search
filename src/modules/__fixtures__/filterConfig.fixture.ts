import { 
    buildFilterConfigData, 
    Filter, 
    FilterConfig,
    FiltersByGroup,
    FilterToGroup,
} from '../filterConfiguration/filterConfig.model';

import { Operators } from '../filterConfiguration/operators';

export const filterDictionary: Dictionary<Filter> = {
    priceMin: { id: 'priceMin', field: 'price', operator: Operators.gt, operand: 200 },
    priceMax: { id: 'priceMax', field: 'price', operator: Operators.lt, operand: 500 },
    numberOfPeople: { id: 'numberOfPeople', field: 'numberOfPeople', operator: Operators.equals, operand: 2 },
    "activity-1": { id: 'activity-1', field: 'activity', operator: Operators.contains, operand: 'swimming' },
    "activity-2": { id: 'activity-2', field: 'activity', operator: Operators.contains, operand: 'tennis' },
    "activity-3": { id: 'activity-3', field: 'activity', operator: Operators.contains, operand: 'golfing' },
};

export const filterByGroup: FiltersByGroup = {
    '0': [filterDictionary.priceMin],
    '1': [filterDictionary.priceMax],
    '2': [filterDictionary.numberOfPeople],
    '3': [filterDictionary['activity-1'], filterDictionary['activity-2'], filterDictionary['activity-3']],
}

export const filterToGroup: FilterToGroup = new Map([
    [filterDictionary.priceMin.id, '0'],
    [filterDictionary.priceMax.id, '1'],
    [filterDictionary.numberOfPeople.id, '2'],
    [filterDictionary['activity-1'].id, '3'],
    [filterDictionary['activity-2'].id, '3'],
    [filterDictionary['activity-3'].id, '3'],
]);

export const filterConfig: FilterConfig = [
    [filterDictionary.priceMin],
    [filterDictionary.priceMax],
    [filterDictionary.numberOfPeople],
    [filterDictionary['activity-1'], filterDictionary['activity-2'], filterDictionary['activity-3']]
];

export const filtersApplied = [filterDictionary.priceMin, filterDictionary['activity-1'], filterDictionary['activity-2']];
export const filtersIdsApplied = ['priceMin', 'activity-1', 'activity-2'];
export const filterConfigData = buildFilterConfigData(filterConfig)(filtersIdsApplied);
