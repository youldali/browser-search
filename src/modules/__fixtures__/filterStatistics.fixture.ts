import { filterConfigData, filterDictionary } from './filterConfig.fixture';
import { itemToFilteringStatus } from './filteringStatus.fixture';
import { createFilteringStatistics, FilterIdToMatchingItemIds } from '../filteringStatistics';


export const filterIdToMatchingItemIds: FilterIdToMatchingItemIds = {
    [filterDictionary.priceMin.id]: [0, 1, 3],
    [filterDictionary.priceMax.id]: [0, 2, 4],
    [filterDictionary.numberOfPeople.id]: [1, 4],
    [filterDictionary['activity-1'].id]: [0, 1, 2],
    [filterDictionary['activity-2'].id]: [0, 1, 2],
    [filterDictionary['activity-3'].id]: [1, 3],
}

const filteringStatisticsDataBuilder = createFilteringStatistics(filterConfigData)(filterIdToMatchingItemIds);
itemToFilteringStatus.forEach( (filteredItemStatus, item) => {
        filteringStatisticsDataBuilder.addFilteredObjectStatus(filteredItemStatus, item.id)
    }
);
export const filteringStatisticsData = filteringStatisticsDataBuilder.done();
