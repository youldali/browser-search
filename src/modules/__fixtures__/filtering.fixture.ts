import { filterConfigData, filterDictionary } from './filterConfig.fixture'
import { operatorToFunction } from '../filterConfiguration/operators';
import { 
    FilterFunctionsCollection,
    FilterGroupToFilterFunctions,
    FilterFunctionsToFilterGroup,
    getFilteringData,
} from '../filtering/filtering.model';

const priceMinFilterFunction = (target: any) => 
    operatorToFunction[filterDictionary.priceMin.operator](target?.[filterDictionary.priceMin.field], filterDictionary.priceMin.operand);

const activity1FilterFunction = (target: any) => 
    operatorToFunction[filterDictionary['activity-1'].operator](target?.[filterDictionary['activity-1'].field], filterDictionary['activity-1'].operand);

const activity2FilterFunction = (target: any) => 
    operatorToFunction[filterDictionary['activity-2'].operator](target?.[filterDictionary['activity-2'].field], filterDictionary['activity-2'].operand);

export const filterFunctionsCollection: FilterFunctionsCollection = [
    [priceMinFilterFunction],
    [activity1FilterFunction, activity2FilterFunction]
];

export const filterGroupToFilterFunctions: FilterGroupToFilterFunctions = {
    '0': filterFunctionsCollection[0],
    '3': filterFunctionsCollection[1],
}

export const filterFunctionsToFilterGroup: FilterFunctionsToFilterGroup = new Map([
    [filterGroupToFilterFunctions['0'], '0'],
    [filterGroupToFilterFunctions['3'], '3'],
]);

export const sampleData = [
    { price: 50, numberOfPeople: 1, activity: ['swimming'] },
    { price: 550, numberOfPeople: 2, activity: ['swimming', 'tennis'] },
    { price: 200, numberOfPeople: 1, activity: [] },
];

export const filteringData = getFilteringData(filterConfigData);