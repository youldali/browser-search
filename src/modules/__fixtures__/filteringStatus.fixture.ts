import { getFilterStatusForItem, FilteredBoxStatus } from '../filtering';
import { filteringData } from './filtering.fixture';

export const items: any[] = [
    { id: 0, price: 250, numberOfPeople: 1, activity: ['swimming', 'tennis']},
    { id: 1, price: 1000, numberOfPeople: 2, activity: ['football', 'swimming', 'tennis', 'golfing']},
    { id: 2, price: 20, numberOfPeople: 5, activity: ['swimming', 'tennis'] },
    { id: 3, price: 1000, numberOfPeople: 1, activity: ['football', 'golfing']},
    { id: 4, price: 10, numberOfPeople: 2, activity: ['football']},
];

export const itemToFilteringStatus:  Map<any, FilteredBoxStatus> = new Map (
    items.map( item => [
        item,
        getFilterStatusForItem(filteringData)(item),
    ])
);
