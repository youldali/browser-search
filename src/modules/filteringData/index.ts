export {
    FilterIdToMatchingDocumentIds,
    NextFilterState,
    FilteringData,
    createFilteringData,
} from './filteringData.model';

export {
    serialize,
    deserialize,
} from './serialize';