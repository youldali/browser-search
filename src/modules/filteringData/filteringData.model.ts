import { FilterGroupId, FilterId } from 'modules/filterConfiguration'
import { FilteredItemStatus } from 'modules/filteringStatus'
import { FilterConfigData } from 'modules/filterConfiguration'
import { findIntersectionOfSortedArrays } from 'helpers/array.util'

type ItemId = string | number;

export interface FilterIdToMatchingItemIds {
    [key: string]: ItemId[];
}

export type FilteringStat = {
    type: 'added' | 'narrowed',
    itemIds: ItemId[]
}

export interface FilteringData {
    getItemsIdsModifiedByFilter: (filterId: FilterId) => FilteringStat,
    getItemsIdsRejectedByOneGroup: (groupId: FilterGroupId) => ItemId[],
    getItemsIdsRejectedByMultipleFilters: () => ItemId[],
    getItemsIdsValidated: () => ItemId[],
}

type ItemIdsByFilteringStatus = Map<FilterGroupId | boolean, ItemId[]>;

export const createFilteringData = 
    (filterConfigData: FilterConfigData) =>
    (filterIdToMatchingItemIds: FilterIdToMatchingItemIds) => {
        const initMapStructure = (): ItemIdsByFilteringStatus => {
            const itemIdsByFilteringStatus: ItemIdsByFilteringStatus = new Map();
            itemIdsByFilteringStatus
                .set(true, [])
                .set(false, []);

            return itemIdsByFilteringStatus;
        };

        const addItemIdToBoolean = (hasPassed: boolean, id: ItemId) => {
            const itemIds = itemIdsByFilteringStatus.get(hasPassed) as ItemId[];
            itemIds.push(id);      
        };

        const addItemIdToRejectedGroup = (group: FilterGroupId, id: ItemId) => {
            const listForGroup = itemIdsByFilteringStatus.get(group);
            listForGroup 
                ? listForGroup.push(id)
                : itemIdsByFilteringStatus.set(group, [id]);
        };

        const itemIdsByFilteringStatus = initMapStructure();

        return {
            addFilteredObjectStatus(filteredItemStatus: FilteredItemStatus, itemId: ItemId) {
                return (
                    filteredItemStatus.pass 
                        ? addItemIdToBoolean(true, itemId) :
                    filteredItemStatus.filterGroupRejected 
                        ? addItemIdToRejectedGroup(filteredItemStatus.filterGroupRejected, itemId) 
                        : addItemIdToBoolean(false, itemId)
                );
            },

            setStatusValue(filteredItemStatus: FilteredItemStatus, idList: ItemId[]) {
                filteredItemStatus.filterGroupRejected 
                    ? itemIdsByFilteringStatus.set(filteredItemStatus.filterGroupRejected, idList) 
                    : itemIdsByFilteringStatus.set(filteredItemStatus.pass, idList);

                return this;
            },

            done(): FilteringData {
                return getFilteringData(filterConfigData)(filterIdToMatchingItemIds)(itemIdsByFilteringStatus);
            }
        };
    };


const getFilteringData = 
    (filterConfigData: FilterConfigData) =>
    (filterIdToMatchingItemIds: FilterIdToMatchingItemIds) =>
    (itemIdsByFilteringStatus: ItemIdsByFilteringStatus): FilteringData => {

        /**
         * If 1 filter of the group has been checked, we are in this case:
         * the filter will actually add some new items since the logical operation is an OR with the other filters in the group
         */
        const getItemIdsAddedByFilter = (filterId: FilterId, filterGroupId: FilterGroupId): FilteringStat => {
            const itemsIdsRejectedByFilterGroup = itemIdsByFilteringStatus.get(filterGroupId) as ItemId[];
            const itemIdsMatchingFilterId = filterIdToMatchingItemIds[filterId]; 
            const itemIdsAddedIfFilterIsChecked = findIntersectionOfSortedArrays(itemIdsMatchingFilterId)(itemsIdsRejectedByFilterGroup);

            return {
                type: 'added',
                itemIds: itemIdsAddedIfFilterIsChecked,
            }
        }

        /**
         * If 0 filter of the group has been checked, we are in this case:
         * the filter will narrow the list of items since the logical operation is an AND with the rest of the filters
         */
        const getItemIdsNarrowedByFilter = (filterId: FilterId): FilteringStat => {
            const itemIdsMatchingFilterId = filterIdToMatchingItemIds[filterId];
            const itemIdsValidated = itemIdsByFilteringStatus.get(true) as ItemId[];
            const itemIdsLeftIfFilterIsChecked = findIntersectionOfSortedArrays(itemIdsMatchingFilterId)(itemIdsValidated);

            return {
                type: 'narrowed',
                itemIds: itemIdsLeftIfFilterIsChecked,
            }
        }


        const getItemsIdsModifiedByFilter = (filterId: FilterId): FilteringStat => {
            const filterGroupId = filterConfigData.getGroupIdForFilter(filterId);
            return itemIdsByFilteringStatus.get(filterGroupId) === undefined 
                ? getItemIdsNarrowedByFilter(filterId)
                : getItemIdsAddedByFilter(filterId, filterGroupId)
        }

        return {
            getItemsIdsModifiedByFilter,
            getItemsIdsRejectedByOneGroup: (filterGroupId: FilterGroupId) => itemIdsByFilteringStatus.get(filterGroupId) ?? [],
            getItemsIdsRejectedByMultipleFilters: () => itemIdsByFilteringStatus.get(false) ?? [],
            getItemsIdsValidated: () => itemIdsByFilteringStatus.get(true) ?? [],
        }
    }
