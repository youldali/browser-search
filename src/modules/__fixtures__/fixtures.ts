import { 
    buildFilterConfigData, 
    Filter, 
    FilterConfig,
    FiltersByGroup,
} from '../filterConfiguration/filterConfig.model';
import { Operators, operatorToFunction } from '../filterConfiguration/operators';
import { 
    FilterFunctionsCollection,
    FilterGroupToFilterFunctions,
    FilterFunctionsToFilterGroup,
    FilteredItemStatus,
    getFilteringFunctionsData,
    getFilterStatusForItem,
} from '../filteringStatus';
import { createFilteringData, FilterIdToMatchingItemIds } from '../filteringData';


export const items: any[] = [
    { id: 0, price: 250, numberOfPeople: 1, activity: ['swimming', 'tennis']},
    { id: 1, price: 1000, numberOfPeople: 2, activity: ['football', 'swimming', 'tennis', 'golfing']},
    { id: 2, price: 20, numberOfPeople: 5, activity: ['swimming', 'tennis'] },
    { id: 3, price: 1000, numberOfPeople: 1, activity: ['football', 'golfing']},
    { id: 4, price: 10, numberOfPeople: 2, activity: ['football']},
];

/**
 * FilterConfiguration Module
 */
export const filterDictionary: Dictionary<Filter> = {
    priceMin: { id: 'priceMin', field: 'price', operator: Operators.gt, operand: 200 },
    priceMax: { id: 'priceMax', field: 'price', operator: Operators.lt, operand: 500 },
    numberOfPeople: { id: 'numberOfPeople', field: 'numberOfPeople', operator: Operators.equals, operand: 2 },
    "activity-1": { id: 'activity-1', field: 'activity', operator: Operators.contains, operand: 'swimming' },
    "activity-2": { id: 'activity-2', field: 'activity', operator: Operators.contains, operand: 'tennis' },
    "activity-3": { id: 'activity-3', field: 'activity', operator: Operators.contains, operand: 'golfing' },
};

export const filterByGroup: FiltersByGroup = {
    '0': [filterDictionary.priceMin],
    '1': [filterDictionary.priceMax],
    '2': [filterDictionary.numberOfPeople],
    '3': [filterDictionary['activity-1'], filterDictionary['activity-2'], filterDictionary['activity-3']],
}

export const filterConfig: FilterConfig = [
    [filterDictionary.priceMin],
    [filterDictionary.priceMax],
    [filterDictionary.numberOfPeople],
    [filterDictionary['activity-1'], filterDictionary['activity-2'], filterDictionary['activity-3']]
];

export const filtersIdsApplied = ['priceMin', 'activity-1', 'activity-2'];
export const filterConfigData = buildFilterConfigData(filterConfig)(filtersIdsApplied);

/**
 * Filtering Module
 */

const priceMinFilterFunction = (target: any) => 
    operatorToFunction[filterDictionary.priceMin.operator](target?.[filterDictionary.priceMin.field], filterDictionary.priceMin.operand);

const activity1FilterFunction = (target: any) => 
    operatorToFunction[filterDictionary['activity-1'].operator](target?.[filterDictionary['activity-1'].field], filterDictionary['activity-1'].operand);

const activity2FilterFunction = (target: any) => 
    operatorToFunction[filterDictionary['activity-2'].operator](target?.[filterDictionary['activity-2'].field], filterDictionary['activity-2'].operand);

export const filterFunctionsCollection: FilterFunctionsCollection = [
    [priceMinFilterFunction],
    [activity1FilterFunction, activity2FilterFunction]
];

export const filterGroupToFilterFunctions: FilterGroupToFilterFunctions = {
    '0': filterFunctionsCollection[0],
    '3': filterFunctionsCollection[1],
}

export const filterFunctionsToFilterGroup: FilterFunctionsToFilterGroup = new Map([
    [filterGroupToFilterFunctions['0'], '0'],
    [filterGroupToFilterFunctions['3'], '3'],
]);

export const filteringData = getFilteringFunctionsData(filterConfigData);

export const itemToFilteringStatus:  Map<any, FilteredItemStatus> = new Map (
    items.map( item => [
        item,
        getFilterStatusForItem(filteringData)(item),
    ])
);


/**
 * FilterStatistics Module
 */

export const filterIdToMatchingItemIds: FilterIdToMatchingItemIds = {
    [filterDictionary.priceMin.id]: [0, 1, 3],
    [filterDictionary.priceMax.id]: [0, 2, 4],
    [filterDictionary.numberOfPeople.id]: [1, 4],
    [filterDictionary['activity-1'].id]: [0, 1, 2],
    [filterDictionary['activity-2'].id]: [0, 1, 2],
    [filterDictionary['activity-3'].id]: [1, 3],
}

const filteringStatisticsDataBuilder = createFilteringData(filterConfigData)(filterIdToMatchingItemIds);
itemToFilteringStatus.forEach( (filteredItemStatus, item) => {
        filteringStatisticsDataBuilder.addFilteredObjectStatus(filteredItemStatus, item.id)
    }
);
export const filteringStatisticsData = filteringStatisticsDataBuilder.done();