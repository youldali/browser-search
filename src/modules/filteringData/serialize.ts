import { DocumentId, FilteringData, NextFilterState } from './filteringData.model'
import { GroupId, FilterId, FilterConfigData } from 'modules/filterConfiguration'

export interface SerializedFilteringData {
  documentsIdsValidated: DocumentId[];
  documentsIdsRejectedByMultipleFilters: DocumentId[];
  nextFilterStatesForNonAppliedFilterId: Dictionary<NextFilterState>;
  documentsIdsRejectedByGroupId: Dictionary<DocumentId[]>;
  nextFilterStateForFilterId: Dictionary<NextFilterState>;
}

export const serialize = <T>(filterConfigData: FilterConfigData<T>) => (filteringData: FilteringData): SerializedFilteringData => {
  const documentsIdsValidated = filteringData.getDocumentsIdsValidated();
  const documentsIdsRejectedByMultipleFilters = filteringData.getDocumentsIdsRejectedByMultipleFilters();
  const nextFilterStatesForNonAppliedFilterId = filteringData.getNextFilterStatesForNonAppliedFilterId();

  const groupIds = filterConfigData.getAllFilterGroupIds();
  const documentsIdsRejectedByGroupId = groupIds.reduce((acc: Dictionary<DocumentId[]>, groupId: GroupId): Dictionary<DocumentId[]> => {
    acc[groupId] = filteringData.getDocumentsIdsRejectedByGroupId(groupId);
    return acc;
  }, {})

  const filterIds = filterConfigData.getAllFilterIds();
  const nextFilterStateForFilterId = filterIds.reduce((acc: Dictionary<NextFilterState>, filterId: FilterId): Dictionary<NextFilterState> => {
    acc[filterId] = filteringData.getNextFilterStateForFilterId(filterId);
    return acc;
  }, {})

  return {
    documentsIdsValidated,
    documentsIdsRejectedByMultipleFilters,
    nextFilterStatesForNonAppliedFilterId,
    documentsIdsRejectedByGroupId, 
    nextFilterStateForFilterId
  }
}

export const deserialize = (serializedFilteringData: SerializedFilteringData): FilteringData => {
  const getDocumentsIdsValidated = () => serializedFilteringData.documentsIdsValidated;
  const getDocumentsIdsRejectedByMultipleFilters = () => serializedFilteringData.documentsIdsRejectedByMultipleFilters;
  const getNextFilterStatesForNonAppliedFilterId = () => serializedFilteringData.nextFilterStatesForNonAppliedFilterId;
  const getNextFilterStateForFilterId = (filterId: FilterId) => (serializedFilteringData.nextFilterStateForFilterId[filterId]);
  const getDocumentsIdsRejectedByGroupId = (groupId: GroupId) => (serializedFilteringData.documentsIdsRejectedByGroupId[groupId]);

  return {
    getNextFilterStatesForNonAppliedFilterId,
    getNextFilterStateForFilterId,
    getDocumentsIdsRejectedByGroupId,
    getDocumentsIdsRejectedByMultipleFilters,
    getDocumentsIdsValidated,
  }

}