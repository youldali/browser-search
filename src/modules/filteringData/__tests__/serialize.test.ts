import { createFilteringData } from '../filteringData.model';
import { serialize, deserialize } from '../serialize';
import { filterConfigDataFixture, filterIdToMatchingDocumentIdsFixture, itemToFilteringStatusFixture } from './__fixtures__/fixtures';

describe ('serialization', () => {
    const filteringStatisticsDataBuilder = createFilteringData(filterConfigDataFixture)(filterIdToMatchingDocumentIdsFixture);
    itemToFilteringStatusFixture.forEach( (filteredDocumenttatus, item) => {
            filteringStatisticsDataBuilder.addFilteredObjectStatus(filteredDocumenttatus, item.id)
        }
    );
    const filteringStatisticsData = filteringStatisticsDataBuilder.done();

    describe('serialize', () => {
        test('it should serialize the "FilteringData" data type', () => {
        expect(serialize(filterConfigDataFixture)(filteringStatisticsData)).toMatchSnapshot();
      });
    });
    
    describe('deserialize', () => {
        const deserializeFilteringData = deserialize(serialize(filterConfigDataFixture)(filteringStatisticsData));
    
        test('Should give all the document id validated', () => {
            expect(deserializeFilteringData.getDocumentsIdsValidated()).toEqual(filteringStatisticsData.getDocumentsIdsValidated());
        });
        
        test('Should give all the document id rejected by multiple filters', () => {
            expect(deserializeFilteringData.getDocumentsIdsRejectedByMultipleFilters()).toEqual(filteringStatisticsData.getDocumentsIdsRejectedByMultipleFilters());
        });
    
        test('Should give a dictionary of FilteringStat by filter non applied', () => {
            expect(deserializeFilteringData.getNextFilterStatesForNonAppliedFilterId()).toEqual(filteringStatisticsData.getNextFilterStatesForNonAppliedFilterId());
        });
        
        filterConfigDataFixture.getAllFilterGroupIds().forEach(filterGroupId => {
            test(`Should give all the document id rejected by group "${filterGroupId}"`, () => {
                expect(deserializeFilteringData.getDocumentsIdsRejectedByGroupId(filterGroupId)).toEqual(filteringStatisticsData.getDocumentsIdsRejectedByGroupId(filterGroupId));
            });
        });
    
        filterConfigDataFixture.getFilterIdsNotApplied().forEach(filterId => {
            test(`Should give all the FilteringStat by filter "${filterId}"`, () => {
                expect(deserializeFilteringData.getNextFilterStateForFilterId(filterId)).toEqual(filteringStatisticsData.getNextFilterStateForFilterId(filterId));
            });
        });
    });    

})
