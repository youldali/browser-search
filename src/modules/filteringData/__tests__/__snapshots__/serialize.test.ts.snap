// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serialization serialize it should serialize the "FilteringData" data type 1`] = `
Object {
  "documentsIdsRejectedByGroupId": Object {
    "0": Array [
      2,
    ],
    "1": Array [],
    "2": Array [],
    "3": Array [
      3,
    ],
  },
  "documentsIdsRejectedByMultipleFilters": Array [
    4,
  ],
  "documentsIdsValidated": Array [
    0,
    1,
  ],
  "nextFilterStateForFilterId": Object {
    "activity-1": Object {
      "documentIds": Array [],
      "type": "added",
    },
    "activity-2": Object {
      "documentIds": Array [],
      "type": "added",
    },
    "activity-3": Object {
      "documentIds": Array [
        3,
      ],
      "type": "added",
    },
    "numberOfPeople": Object {
      "documentIds": Array [
        1,
      ],
      "type": "narrowed",
    },
    "priceMax": Object {
      "documentIds": Array [
        0,
      ],
      "type": "narrowed",
    },
    "priceMin": Object {
      "documentIds": Array [],
      "type": "added",
    },
  },
  "nextFilterStatesForNonAppliedFilterId": Object {
    "activity-3": Object {
      "documentIds": Array [
        3,
      ],
      "type": "added",
    },
    "numberOfPeople": Object {
      "documentIds": Array [
        1,
      ],
      "type": "narrowed",
    },
    "priceMax": Object {
      "documentIds": Array [
        0,
      ],
      "type": "narrowed",
    },
  },
}
`;
